### PR TITLE
Specify semantics of fontSize in 'em' on region

### LINKED ISSUE #206: Use of 'em' units in tts:fontSize on region element is not well defined.
§8.2.9 needs to place a constraint on the use of 'em' units when expressing tts:fontSize on a region element; alternatively, it could specify that 1em is equal to the initial value of tts:fontSize, namely, 1c, for the purpose of resolving an 'em' unit based font size

### PR DESCRIPTION
**Pull request by palemieux**
Close #206

### CODE REVIEW

File: spec/ttml1.xml [hunk 1]
```diff
@@ -4303,6 +4303,7 @@ glyphs do not have an EM square that is distinct from their font's EM square.</p
 <p>If horizontal and vertical sizes are expressed independently, then the
 units of the <loc href="#style-value-length">&lt;length&gt;</loc> values must be the same.</p>
 <p>The <loc href="#style-value-length">&lt;length&gt;</loc> value(s) used to express font size must be non-negative.</p>
+<p><loc href="#style-value-length">&lt;length&gt;</loc> values specified on a <loc href="#content-vocabulary-span"><el>region</el></loc> element using <code>em</code> units are considered relative to the <emph>Computed Cell Size</emph>.</p>
```
Comment (nigelmegitt) [inline spec/ttml1.xml]: Are they independently related to the cell height and width, or always to the cell height? For example, is `"1em 1em"` supposed to result in the same behaviour as `"100% 100%"` or as `"100%"`?

**Review by nigelmegitt — COMMENTED** [review]
Question asked

File: spec/ttml1.xml (hunk 1 above)
Comment (palemieux) [reply spec/ttml1.xml]: I think this is covered by the note:

_The expression 1c means one cell, where 'c' expresses the cell length unit as defined by 8.3.9 <length>. When a single <length> is expressed using cell units, then it refers to the height of the Computed Cell Size. When two <length> values are expressed using cell units, then the first refers to the width of the Computed Cell Size, and the second refers to the height of the Computed Cell Size._

Comment (nigelmegitt) [reply spec/ttml1.xml]: @palemieux that note explicitly refers to the use of cell units not `em` units so I don't think it does cover it.

Comment (nigelmegitt) [reply spec/ttml1.xml]: By the way, there's another relevant paragraph at https://www.w3.org/TR/ttml1/#style-value-length :

> When specified relative to a font whose size is expressed as a single length measure or as two length measures of equal length, the unit of measure em is considered to be identical to that defined by [XSL 1.1], § 5.9.13; however, when specified relative to a font whose size is expressed as two length measures of non-equal lengths, then one em is equal to the inline progression dimension of the anamorphically scaled font when used to specify lengths in the inline progression direction and equal to the block progression dimension of the scaled font when used to specify lengths in the block progression direction.

However this doesn't really help when the words "relative to a font" don't apply, as here, where the `em` measure is relative to the computed cell size.

**Comment by palemieux** [comment]
@nigelmegitt See text at https://github.com/w3c/ttml1/pull/279/commits/51aa74f4ea3e2cfdf4a3adbf4f8843c1fcb8c72c

File: spec/ttml1.xml [hunk 2]
```diff
@@ -4303,7 +4303,12 @@ glyphs do not have an EM square that is distinct from their font's EM square.</p
 <p>If horizontal and vertical sizes are expressed independently, then the
 units of the <loc href="#style-value-length">&lt;length&gt;</loc> values must be the same.</p>
 <p>The <loc href="#style-value-length">&lt;length&gt;</loc> value(s) used to express font size must be non-negative.</p>
-<p><loc href="#style-value-length">&lt;length&gt;</loc> values specified on a <loc href="#content-vocabulary-span"><el>region</el></loc> element using <code>em</code> units are considered relative to the <emph>Computed Cell Size</emph>.</p>
+<p><loc href="#style-value-length">&lt;length&gt;</loc> values specified on a <loc href="#content-vocabulary-span">
+<el>region</el></loc> element using <code>em</code>
```
Comment (nigelmegitt) [inline spec/ttml1.xml]: the link points to span but the text says region - copy/paste error?

**Review by nigelmegitt — CHANGES_REQUESTED** [review]
Thanks, the update addresses the comment, just noticed a typo to be fixed before approval.

**Comment by palemieux** [comment]
@nigelmegitt Turns out the paste was correct, but the spec was incorrect where the copy was made :) Fixed now.

**Review by nigelmegitt — APPROVED** [review]
Thanks. Funny way to spot spec bugs, but whatever works!

File: spec/ttml1.xml [hunk 3]
```diff
@@ -4303,6 +4303,12 @@ glyphs do not have an EM square that is distinct from their font's EM square.</p
 <p>If horizontal and vertical sizes are expressed independently, then the
 units of the <loc href="#style-value-length">&lt;length&gt;</loc> values must be the same.</p>
 <p>The <loc href="#style-value-length">&lt;length&gt;</loc> value(s) used to express font size must be non-negative.</p>
+<p><loc href="#style-value-length">&lt;length&gt;</loc> values specified on a <loc href="#layout-vocabulary-region">
+<el>region</el></loc> element using <code>em</code> 
+units are considered relative to the <emph>Computed Cell Size</emph>. When a single value is specified, then it is
```
Comment (skynavga) [inline spec/ttml1.xml]: Is the second sentence of this paragraph "When a single ..." dependent on the context in the first sentence, i.e., does the second sentence only apply for length's using ``em``? Or is a general statement (all lengths)? If the former, then change the second sentence to start with "In that case, when a single ...".

File: spec/ttml1.xml [hunk 4]
```diff
@@ -4303,6 +4303,12 @@ glyphs do not have an EM square that is distinct from their font's EM square.</p
 <p>If horizontal and vertical sizes are expressed independently, then the
 units of the <loc href="#style-value-length">&lt;length&gt;</loc> values must be the same.</p>
 <p>The <loc href="#style-value-length">&lt;length&gt;</loc> value(s) used to express font size must be non-negative.</p>
+<p><loc href="#style-value-length">&lt;length&gt;</loc> values specified on a <loc href="#layout-vocabulary-region">
```
Comment (skynavga) [inline spec/ttml1.xml]: Need to tweak to prevent use of ``<length>`` at start of sentence, e.g., change to "When ``<length>`` values ...".

**Review by skynavga — CHANGES_REQUESTED** [review]
See above comments around lines 4306-4308.

**Comment by palemieux** [comment]
@skynavga Edits made

File: spec/ttml1.xml [hunk 5]
```diff
@@ -4303,6 +4303,14 @@ glyphs do not have an EM square that is distinct from their font's EM square.</p
 <p>If horizontal and vertical sizes are expressed independently, then the
 units of the <loc href="#style-value-length">&lt;length&gt;</loc> values must be the same.</p>
 <p>The <loc href="#style-value-length">&lt;length&gt;</loc> value(s) used to express font size must be non-negative.</p>
+<p>When <loc href="#style-value-length">&lt;length&gt;</loc> values is specified on a <loc href="#layout-vocabulary-region">
```
Comment (skynavga) [inline spec/ttml1.xml]: Needs a final tweak, either "When a ``<length>`` value is ..." or "When ``<length>`` values are ...".

**Review by skynavga — APPROVED** [review]
Thanks.

**Review by nigelmegitt — APPROVED** [review]

**Comment by palemieux** [comment]
Propose to replace by #301, which is based on TTML2 prose

**Comment by palemieux** [comment]
Mooted by merging of #301

**Pull request closed without merging**